lib/invitations/inviteUserByEmail.ts

```diff
@@ -0,0 +1,208 @@
+import { FormRecord, UserAbility } from "@lib/types";
+import { getUser } from "@lib/users";
+import {
+  InvalidDomainError,
+  MismatchedEmailDomainError,
+  TemplateNotFoundError,
+  UserAlreadyHasAccessError,
+  UserNotFoundError,
+} from "./exceptions";
+import { getTemplateWithAssociatedUsers } from "@lib/templates";
+import { prisma } from "@lib/integration/prismaConnector";
+import { sendEmail } from "@lib/integration/notifyConnector";
+import { inviteToCollaborate } from "@lib/invitations/emailTemplates/inviteToCollaborate";
+import { inviteToForms } from "@lib/invitations/emailTemplates/inviteToForms";
+import { getOrigin } from "@lib/origin";
+import { logMessage } from "@lib/logger";
+import { Invitation } from "@prisma/client";
+import { logEvent } from "@lib/auditLogs";
+import { isValidGovEmail } from "@lib/validation/validation";
+
+/**
+ * Invite someone to the form by email
+ *
+ * @param ability
+ * @param email
+ * @param formId
+ */
+export const inviteUserByEmail = async (
+  ability: UserAbility,
+  email: string,
+  formId: string,
+  message: string
+) => {
+  let invitation;
```

Comment:
We may want to add a type to `invitation` since it's known and we're already importing the type.